aiida/cmdline/commands/cmd_calcjob.py

```diff
@@ -90,20 +90,25 @@ def calcjob_inputcat(calcjob, path):
     """
     from aiida.plugins.entry_point import get_entry_point_from_class
 
+    # Get path from the given CalcJobNode if not defined by user
     if path is None:
-
         path = calcjob.get_option('input_filename')
 
-        if path is None:
-            cls = calcjob.__class__
-            _, entry_point = get_entry_point_from_class(cls.__module__, cls.__name__)
-            echo.echo_critical('{} does not define a default input file. Please specify a path explicitly'.format(
-                entry_point.name))
+    # Get path from current process class of CalcJobNode if still not defined
+    if path is None:
+        fname = calcjob.options.get('input_filename')
```

Comment:
wow, so `calcjob.get_option` returns something different than `calcjob.options.get`?
@sphuber I think that is potentially very confusing. Should I open an issue for this?

Reply:
To me it makes sense if the goal is:
Show me all the options I can set: `calcjob.options`
Give me options that have been set for the specific node: `calcjob.get_option('x')`.

But yeah, the naming is a bit confusing, I agree, even if the above is the intended way it should work.

Reply:
The problem is exactly that. In principle, only `get_option` should exist, i.e. what was the value of this option that was set on this `CalcJobNode`. The `options` property is just a shortcut to loading the `Process` class used in generating the `CalcJobNode` and inspecting the `metadata.options` port from its `ProcessSpec`. Maybe we should remove this shortcut, such that *if* people use it, they realize damn well that it concerns the options of a spec, not option values that were actually set. Removing the shortcut would actually not make the alternative all that bad even. The current implementation of `CalcJobNode.options` is:
```
    @property
    def options(self):
        """Return the available process options for the process class that created this node."""
        try:
            return self.process_class.spec().inputs._ports['metadata']['options']  # pylint: disable=protected-access
        except ValueError:
            return {}
```
But this can actually be simplified a lot to `self.process_class.options` since I already added the `options` property to the `Process` class some time ago. So my proposal would simply be to remove `options` from the `CalcJobNode` class such that people have to do `self.process_class.options` which should hopefully make it clear that these are the abstract options

Reply:
I think this would be a good solution.